File: app/src/main/java/com/tomaszrykala/dorefindmi/things/supplier/abcbuttons/AbcButtonsSupplier.java
package com.tomaszrykala.dorefindmi.things.supplier.abcbuttons;

import com.google.android.things.contrib.driver.button.Button;
import com.tomaszrykala.dorefindmi.model.AbcButton;
import com.tomaszrykala.dorefindmi.things.supplier.BaseSupplier;

public interface AbcButtonsSupplier extends BaseSupplier, Button.OnButtonEventListener {
    void setListener(Listener listener);

    interface Listener {
        void onButtonEvent(AbcButton abcButton, boolean pressed);
    }
}

File: app/src/main/java/com/tomaszrykala/dorefindmi/things/supplier/abcbuttons/AbcButtonsSupplier.kt
package com.tomaszrykala.dorefindmi.things.supplier.abcbuttons

import com.google.android.things.contrib.driver.button.Button
import com.tomaszrykala.dorefindmi.domain.AbcButton
import com.tomaszrykala.dorefindmi.things.supplier.BaseSupplier

interface AbcButtonsSupplier : BaseSupplier, Button.OnButtonEventListener {

    fun setListener(listener: Listener)

    interface Listener {
        fun onButtonEvent(abcButton: AbcButton, pressed: Boolean)
    }
}
